plutus-tx/src/PlutusTx/Builtins/Internal.hs

```diff
@@ -101,6 +114,9 @@ chooseUnit (BuiltinUnit ()) a = a
 INTEGER
 -}
 
+-- I'm somewhat surprised that this works despite not being opaque! GHC doesn't seem to
+-- mess with 'Integer', which is suspicious to me. Probably we *should* make 'BuiltinInteger'
+-- opaque, but that's going to really mess with our users' code...
 type BuiltinInteger = Integer
```

Comment:
perhaps because ghc only tries to optimize ghc-prim datatypes and does not touch integer-gmp/ghc-bignum stuff? :man_shrugging: